handlers/africastalking/handler.go

```diff
@@ -114,21 +114,16 @@ func (h *handler) receiveStatus(ctx context.Context, channel courier.Channel, w
 }
 
 // Send sends the given message, logging any HTTP calls or errors
-func (h *handler) Send(ctx context.Context, msg courier.MsgOut, clog *courier.ChannelLog) (courier.StatusUpdate, error) {
+func (h *handler) Send(ctx context.Context, msg courier.MsgOut, res *courier.SendResult, clog *courier.ChannelLog) error {
 	isSharedStr := msg.Channel().ConfigForKey(configIsShared, false)
 	isShared, _ := isSharedStr.(bool)
 
 	username := msg.Channel().StringConfigForKey(courier.ConfigUsername, "")
-	if username == "" {
-		return nil, fmt.Errorf("no username set for AT channel")
-	}
-
 	apiKey := msg.Channel().StringConfigForKey(courier.ConfigAPIKey, "")
-	if apiKey == "" {
-		return nil, fmt.Errorf("no API key set for AT channel")
-	}
 
-	status := h.Backend().NewStatusUpdate(msg.Channel(), msg.ID(), courier.MsgStatusErrored, clog)
+	if username == "" || apiKey == "" {
+		return courier.ErrSendChannelConfig
+	}
```

Comment:
not so worried about making these errors less specific.. this should only happen if we've messed up channel creation in RP and we'll get this in Sentry.